tests/test_web.py

```diff
@@ -24,8 +24,10 @@ def test_new_user_random(db, monkeypatch, mailcow):
     assert r.json.get("reason") == "token 123123 is invalid"
 
     # delete a@x.testrun.org and b@x.testrun.org in case earlier tests failed to clean them up
-    mailcow.del_user_mailcow("pytest.a@x.testrun.org")
-    mailcow.del_user_mailcow("pytest.b@x.testrun.org")
+    user_a = "pytest.a@" + mailcow_domain
+    user_b = "pytest.b@" + mailcow_domain
+    mailcow.del_user_mailcow(user_a)
+    mailcow.del_user_mailcow(user_b)
```

Comment:
this looks flaky to randomly try to remove users.  Rather, at the place where they are obtained/created, register an invocation of `mailcow.del_user_mailcow`  to run at function finalization/cleanup -- the fixture `request` and `request.addfinalizer()` can be used in a test function.